lib/req/steps.ex

```diff
@@ -1135,15 +1135,11 @@ defmodule Req.Steps do
   ## Utilities
 
   defp put_new_header(struct, name, value) do
```

Comment:
all this function does is calling Req.Request.put_new_header. Let's remove this indirection and call the function directly at call sites